checker/src/org/checkerframework/checker/nullness/NullnessAnalysis.java

```diff
@@ -41,6 +46,17 @@ public NullnessValue createAbstractValue(AnnotatedTypeMirror type) {
             // 'no information'.
             return null;
         }
-        return new NullnessValue(this, type);
+        if (type.getKind() == TypeKind.WILDCARD) {
+            AnnotatedWildcardType wildcard = (AnnotatedWildcardType) type;
+            Set<AnnotationMirror> annos = wildcard.getExtendsBound().getAnnotations();
+            return new NullnessValue(this, annos, type.getUnderlyingType());
+        }
+        return new NullnessValue(this, type.getAnnotations(), type.getUnderlyingType());
```

Comment:
Would using `getEffectiveAnnotations` cover both wildcards and type variables and not require the special case above?